src/smt/smt_engine.cpp

```diff
@@ -3860,7 +3839,8 @@ Result SmtEngine::check() {
        (not d_logic.isQuantified() &&
         d_logic.isPure(THEORY_ARITH) && d_logic.isLinear() && !d_logic.isDifferenceLogic() &&  !d_logic.areIntegersUsed()
         )){
-      if(d_private->d_iteSkolemMap.empty()){
+      if (d_private->d_assertions.getIteSkolemMap().empty())
```

Comment:
Do you think `d_assertions` of `SmtEnginePrivate` will be used in other places? If not, maybe it is a good idea to keep its `d_assertions` private, and have a public `getIteSkolemMap()` whose body is `return d_assertions.getIteSkolemMap()`?

Reply:
This is a good point, I made `d_assertions` private again.